File: backend/src/main/scala/microservices/operations/airline/objects/ManagerFlightOrderResponse.scala
// 这个文件定义 airline 单个订单项的响应。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerFlightOrderResponse(
    orderId: String,
    orderItemId: String,
    buyerUserId: String,
    buyerNickname: String,
    cabinClass: String,
    orderStatus: String,
    orderCreatedAt: String,
    travelerIds: List[String],
    travelers: List[ManagerFlightOrderTravelerResponse]
)
object ManagerFlightOrderResponse:
  given sourceEncoder: Encoder[ManagerFlightOrderResponse] = deriveEncoder
  given sourceDecoder: Decoder[ManagerFlightOrderResponse] = deriveDecoder

File: frontend/src/microservices/operations/airline/objects/ManagerFlightOrderResponse.ts
// 本文件定义 operations 航空管理端的 `ManagerFlightOrderResponse`，作为订单响应数据并提供 JSON 编解码。

import type { ManagerFlightOrderTravelerResponse } from './ManagerFlightOrderTravelerResponse'

export type ManagerFlightOrderResponse = {
  orderId: string
  orderItemId: string
  buyerUserId: string
  buyerNickname: string
  cabinClass: string
  orderStatus: string
  orderCreatedAt: string
  travelerIds: string[]
  travelers: ManagerFlightOrderTravelerResponse[]
}

export const managerFlightOrderResponseFromJson = (json: string): ManagerFlightOrderResponse =>
  JSON.parse(json) as ManagerFlightOrderResponse

export const managerFlightOrderResponseToJson = (value: ManagerFlightOrderResponse): string =>
  JSON.stringify(value)

